Marlin/src/gcode/calibrate/G33.cpp

```diff
@@ -71,7 +71,9 @@ float lcd_probe_pt(const xy_pos_t &xy);
 
 void ac_home() {
   endstops.enable(true);
+  TERN_(SENSORLESS_PROBING, probe.homing_current_on());
   home_delta();
+  TERN_(SENSORLESS_PROBING, probe.homing_current_off());
```

Comment:
This is getting slightly convoluted because "sensorless probing" is substituting for a real probe, so when doing "sensorless probing" we're pretending that the stall is a probe trigger.

When we are homing in this context here we are not doing something related to the probe, as we are when a real probe is being used for Z homing. So telling the "probe" (which does not really exist) that it should set its "homing current" is backwards.

Instead, the firmware should set the "probing current" only when doing a bed probe, and the stepper driver for the Z axis should always fall back to the "homing current" when the probe is not in use.

Reply:
I would suggest that when a "sensorless probing" probe is "deployed" this is when it should set its "probing current" and when it is "stowed" the current should be set back to the default stall value. In a similar manner, when using sensorless _homing_, the current and sensitivity of the axes should be set for homing, and when the procedure is done, they should be set back to their normal print-ready values.

Reply:
> Instead, the firmware should set the "probing current" only when doing a bed probe, and the stepper driver for the Z axis should always fall back to the "homing current" when the probe is not in use.

This part of the code, tries to replicate the same conditions that exist in a homing with "SENSORLESS-HOMING", but in bed probing.

Sorry, it probably has a slightly descriptive name. "probe.homing_current_off ()" aims to return the TMC to normal use current. And that is why this function is used throughout the calibration process. When a HOMING ends and once the nozzle touches the bed in each probe.

Reply:
>I would suggest that when a "sensorless probing" probe is "deployed" this is when it should set its "probing current" and when it is "stowed" the current should be set back to the default print-happy stall value. In a similar manner, when using sensorless homing, the current and sensitivity of the axes should be set for homing, and when the procedure is done, they should be set back to their normal print-ready values.

I agree.
And I have tried to do the same, but also, looking for a place in the code where this procedure is only performed when the nozzle starts the bed probing maneuver, when it starts to go down.

The place I chose was "probe_down_to_z ()"

Reply:
The `home_delta` routine has no relation to the probe, and these `homing_current_on/off` methods are messing with parameters that apply to the X and Y diag pins, not just the diag pin that applies to the probe. So, maybe these calls are okay here, but the methods need to be moved to the `motion.cpp` file and should not be part of the probe class.

Reply:
Hmm… Yes, I see it's a bit of a tangle, as we're trying to use as few static buffers as possible, but also aiming to set and restore various parameters, so it's not possible to localize these things in the delta homing without adding another static buffer. My main issue with this addition is that it is too narrow, referring only to XYZ axes, and also that it has no purpose outside of this single instance of DELTA homing for `G33`. I would ideally like to see less repetition and less use of narrow customizations like this, but admittedly it's a hard thing to accomplish while also minimizing the use of global buffers that are only used for the duration of a single task.

Reply:
> The `home_delta` routine has no relation to the probe, and these `homing_current_on/off` methods are messing with parameters that apply to the X and Y diag pins, not just the diag pin that applies to the probe. So, maybe these calls are okay here, but the methods need to be moved to the `motion.cpp` file and should not be part of the probe class.

In all humility, I disagree. 
`home_delta` is used in G28 and G33. It is certainly a specific method for DELTA (that is why in G28 it is conditioned only to DELTA). In a DELTA any movement is a combination of the three axes, XYZ, either in HOMING or Z probe. For this reason all new methods of my PR always intervene on the three axes equally, and that is the key to success or success. failure in this improvement.

Reply:
> Hmm… Yes, I see it's a bit of a tangle, as we're trying to use as few static buffers as possible, but also aiming to set and restore various parameters, so it's not possible to localize these things in the delta homing without adding another static buffer. My main issue with this addition is that it is too narrow, referring only to XYZ axes, and also that it has no purpose outside of this single instance of DELTA homing for `G33`. I would ideally like to see less repetition and less use of narrow customizations like this, but admittedly it's a hard thing to accomplish while also minimizing the use of global buffers that are only used for the duration of a single task.

It may seem unambitious, but it is that G33 is only used in DELTA that I know of. And that is the objective of this PR since its inception. I mean that a DELTA can use the SENSORLESS_PROBING features just like any other more common printer architecture. That DELTA users are a minority, it should not discriminate against us and that the firmware forget us.
And it certainly only intervenes on printers (DELTA) with XYZ axes. But it is that before this PR, even for DELTA, it only intervened in the Z axis. As I said before, my approach may be not very ambitious, but at least it provides the basic performance for DELTA, and try that the performance in the rest of architectures not affected.

Reply:
> it should not discriminate against us and that the firmware forget us

🙄 Everything in the firmware exists because someone contributed it. If something for delta is lacking, it is only because there are not enough delta users contributing.